### PR TITLE
Log Model Hparams first

### PR DESCRIPTION
**Pull request by nils-werner**
For some reason the [Tensorboard Hparams Screen](https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams) preserves the order of the parameters in the `hparams` dict. That means currently the order of values is

 - Trainer
 - Model
 - Datamodule
 - Seed
 - Callbacks
 - Model/Params

Since most interesting parameters are likely in the model, it is a bit annoying that I always have to scroll down to them. Therefore I would recommend an order like

 - Model
 - Model/Params
 - Datamodule
 - Trainer
 - Seed
 - Callbacks

This PR implements that change.

### CODE REVIEW

**Comment by ashleve** [comment]
Sure. Seems like there's no reason not to do it.
